util/projectRequest.ts

```diff
@@ -11,34 +12,102 @@ export async function getProjects(): Promise<Project[]> {
+  return projects;
+}
+
+//search uclaacm org for good first issues, repos which contain gfis, and match them up
+export async function getGoodFirstIssueProjects(): Promise<GFIProject[]> {
+  const PaginatedOctokit = Octokit.plugin(paginateRest);
+  const octokit = new PaginatedOctokit();
+  const gfiIssuesSearchQuery = 'org:uclaacm+label:"good+first+issue"+is:open';
+  const gfiIssuesResponse = await octokit.paginate('GET /search/issues', {
+    q: gfiIssuesSearchQuery,
+  }) as GitHubIssue[];
+  const gfiReposSearchQuery = 'good-first-issues:>0+org:uclaacm';
+  const gfiReposResponse = await octokit.paginate('GET /search/repositories', {
+    q: gfiReposSearchQuery,
+  }) as GitHubRepo[];
+  const repoMap = new Map<string, GitHubRepo>();
+  //turn the gfiReposResponse into a map of url and the entire repo
+  for (const repo of gfiReposResponse){
+    repoMap.set(repo.url,repo);
+  }
+  const gfiProjects = mapIssuesToProjects(gfiIssuesResponse, repoMap);
+  return gfiProjects;
+}
+
+function mapIssuesToProjects(issues: GitHubIssue[], repoMap: Map<string, GitHubRepo>) : GFIProject[]{
+  //turn the issues into a map of projects and a list of issues associated with it
+  const mappedRepos = new Map<string, GitHubIssue[]>();
+  // console.log(issues);
+  if(issues.length ==0){
+    //console.error('No open good first issues!');
+    return [];
+  }
+  for (const issue of issues){
+    const issueRepoURL = issue.repository_url;
+    //if map already contains that project, append to array of issues, or make a new array containing it
+    mappedRepos.set(issueRepoURL,[...mappedRepos.get(issueRepoURL) ?? [], issue]);
+  }
+
+  const mapIter = mappedRepos.entries();
+
+  //turn everything into a gfiProject[]
+  const gfis : GFIProject[] = [];
+  let itVal = mapIter.next();
+  while(!itVal.done){
+    const [repoUrl, repoIssues] = itVal.value;
+    const correspondingRepo = repoMap.get(repoUrl);
+    if(!correspondingRepo){
+      //console.error('Repo Map went wrong!');
+      return gfis;
+    }
+    const correspondingProject = convertRepoToProject(correspondingRepo);
+    gfis.push({
+      issues: repoIssues,
+      project: correspondingProject,
+      repoURL: repoUrl,
+    });
+    itVal = mapIter.next();
+  }
+  return gfis;
+}
+
+function convertRepoToProject(repo: GitHubRepo): Project {
+  return repo.homepage
+    ? {
+      name: repo.name,
+      description: repo.description ?? '',
+      link: repo.homepage ?? '',
+      repo: repo.html_url,
+      lang: repo.language ?? '',
+      topics: repo.topics ?? [],
+      image: getImageFromTopics(repo.topics).image,
+      alt: getImageFromTopics(repo.topics).alt,
+    }
+    : {
+      name: repo.name,
+      description: repo.description ?? '',
+      repo: repo.html_url ?? '',
+      lang: repo.language ?? '',
+      topics: repo.topics ?? [],
+      image: getImageFromTopics(repo.topics).image,
+      alt: getImageFromTopics(repo.topics).alt,
+    };
+}
```

Comment:
Can't you just conditionally include the key and/or make it nullable/a string? 

In particular, in the positive case for `repo.homepage`, you still add a check for `repo.homepage ?? ''`. Why not just use this in general?

(you can also make this even more concise with destructuring)